Azure Functions/Challenges/AzureFunction.Challenge/Core/Services/OrderService.cs

```diff
@@ -0,0 +1,107 @@
+﻿using AzureFunction.Challenge.Function.Core.Entities;
+using AzureFunction.Challenge.Function.Core.Interfaces;
+using AzureFunction.Challenge.Function.Core.Models;
+using AzureFunction.Challenge.Function.Core.Models.DTOs;
+using Microsoft.EntityFrameworkCore;
+
+namespace AzureFunction.Challenge.Function.Core.Services
+{
+    public class OrderService : IOrderService
+    {
+        private readonly AzureFunctionDbContext _context;
+        public OrderService(AzureFunctionDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<OrderResponseDto> CreateOrder(OrderDto orderDto)
+        {
+            var productIds = orderDto.OrderProducts.Select(op => op.Product.Id).ToList();
+
+            var products = await _context.Products.Where(p => productIds.Contains(p.Id)).ToListAsync();
+
+            var productDictionary = products.ToDictionary(p => p.Id);
+
+            var order = new Order
+            {
+                DeliveryAddress = new Address
+                {
+                    Name = orderDto.DeliveryAddress.Name,
+                    AddressLineOne = orderDto.DeliveryAddress.AddressLineOne,
+                    AddressLineTwo = orderDto.DeliveryAddress.AddressLineTwo,
+                    City = orderDto.DeliveryAddress.City,
+                    PostalCode = orderDto.DeliveryAddress.PostalCode,
+                    Country = orderDto.DeliveryAddress.Country
+                },
+                BillingAddress = new Address
+                {
+                    Name = orderDto.BillingAddress.Name,
+                    AddressLineOne = orderDto.BillingAddress.AddressLineOne,
+                    AddressLineTwo = orderDto.DeliveryAddress.AddressLineTwo,
+                    City = orderDto.BillingAddress.City,
+                    PostalCode = orderDto.BillingAddress.PostalCode,
+                    Country = orderDto.BillingAddress.Country
+                },
+                Email = orderDto.Email,
+                Phone = orderDto.Phone,
+                OrderProducts = orderDto.OrderProducts.Select(op => new OrderProduct
+                {
+                    Quantity = op.Quantity,
+                    Price = productDictionary[op.Product.Id].Price,
+                    Product = productDictionary[op.Product.Id],
+                    Notes = op.Notes
+                }).ToList()
+            };
+
+
+            var entityEntry = await _context.Orders.AddAsync(order);
+            await _context.SaveChangesAsync();
+
+            return new OrderResponseDto
+            {
+                Id = entityEntry.Entity.Id,
+                DeliveryTime = GetDeliveryTime()
```

Comment:
Need to persist DeliveryTime
Add new migration for single column
Save to DB
Return as part of the GetOrder endpoint